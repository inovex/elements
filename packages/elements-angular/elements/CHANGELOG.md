# Changelog

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

## [1.2.0](https://github.com/inovex/elements/compare/v1.1.0...v1.2.0) (2020-10-14)


### Features

* **elements:** add ino-carousel component ([#112](https://github.com/inovex/elements/issues/112)) ([e99b5d7](https://github.com/inovex/elements/commit/e99b5d723903a7060e0691b782d686db01480161))
* **elements|ino-card:** add property to disable hover animation ([#120](https://github.com/inovex/elements/issues/120)) ([0fac301](https://github.com/inovex/elements/commit/0fac301ab4b3fe1b078b5f4d09db97495aa2e779))
* **elements|ino-snackbar:** add property for warning and error states ([#116](https://github.com/inovex/elements/issues/116)) ([b648415](https://github.com/inovex/elements/commit/b64841571322656433ab8ed2ef1439b5cd40c23e))


### Miscellaneous

* hide refactoring commit type in changelog ([#130](https://github.com/inovex/elements/issues/130)) ([00dbf1a](https://github.com/inovex/elements/commit/00dbf1a55828aad4d2c0e29978dc3abe38ab7cbe))

# [1.2.0](https://github.com/inovex/elements/compare/v1.1.0...v1.2.0) (2020-10-14)


### Features

* **elements:** add ino-carousel component ([#112](https://github.com/inovex/elements/issues/112)) ([e99b5d7](https://github.com/inovex/elements/commit/e99b5d723903a7060e0691b782d686db01480161))
* **elements|ino-card:** add property to disable hover animation ([#120](https://github.com/inovex/elements/issues/120)) ([0fac301](https://github.com/inovex/elements/commit/0fac301ab4b3fe1b078b5f4d09db97495aa2e779))
* **elements|ino-snackbar:** add property for warning and error states ([#116](https://github.com/inovex/elements/issues/116)) ([b648415](https://github.com/inovex/elements/commit/b64841571322656433ab8ed2ef1439b5cd40c23e))





# Changelog

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

## [1.1.0](https://github.com/inovex/elements/compare/v1.0.2...v1.1.0) (2020-10-07)


### Features

* **elements|ino-snackbar:** add property to set custom timeout ([#83](https://github.com/inovex/elements/issues/83)) ([d9608da](https://github.com/inovex/elements/commit/d9608daa9779acad4c1f43945a20382ca03f6d4e))


### Miscellaneous

* **elements:** update stencil to v2.0.3 ([#85](https://github.com/inovex/elements/issues/85)) ([99b593d](https://github.com/inovex/elements/commit/99b593d0ee94b8637444a8aa036ebbd6e336490a))


### Refactoring

* **elements:** move from local font files to font package ([#82](https://github.com/inovex/elements/issues/82)) ([c04d903](https://github.com/inovex/elements/commit/c04d903f7f238f950779d9fedf972f56df9647b2))


### Documentation

* **elements-angular:** correct link to top-level readme ([#119](https://github.com/inovex/elements/issues/119)) ([4d57647](https://github.com/inovex/elements/commit/4d576471133ace5b2da67fb98bd77319ad3a504b))

# [1.1.0](https://github.com/inovex/elements/compare/v1.0.2...v1.1.0) (2020-10-07)


### Features

* **elements|ino-snackbar:** add property to set custom timeout ([#83](https://github.com/inovex/elements/issues/83)) ([d9608da](https://github.com/inovex/elements/commit/d9608daa9779acad4c1f43945a20382ca03f6d4e))





# Changelog

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

### [1.0.2](https://github.com/inovex/elements/compare/v1.0.1...v1.0.2) (2020-09-22)

### [1.0.1](https://github.com/inovex/elements/compare/v1.0.0...v1.0.1) (2020-09-22)

## [1.0.0](https://github.com/inovex/elements/compare/v0.21.0...v1.0.0) (2020-09-16)


### âš  BREAKING CHANGES

* We changed the scopes of all our packages from `@inovex` to  `@inovex.de` in order to match our newly created NPM org. Unfortunately, `@inovex` was already taken. To use the inovex-elements beyond version 0.21.0, you have to change the scope in your dependencies from `@inovex/*` to `@inovex.de/*`. This way you will receive our new versions from the public registry of npmjs.
* **elements|ino-button:** The API of the `<ino-button>` has changed. We've replaced the `ino-icon` property.
If you want to use an `ino-icon` inside of a button, use the `ino-icon-leading` or `ino-icon-trailing` property and provide the `<ino-icon>` element as a child of the `<ino-button>` with the slot being either `ino-icon-leading` or `ino-icon-trailing`. See the example below:
    
    ```html
    <!-- Old -->
    <ino-button ino-icon="info">Button with leading icon</ino-button>
    
    <!-- New -->
    <ino-button ino-icon-leading>
      <ino-icon ino-icon="info" slot="ino-icon-leading"></ino-icon>
      Button with leading icon
    </ino-button>
    ```
    
    While this does seem like more work for the consumer, we implemented this change in order to have a consistent way of using the `<ino-icon>` in combination with our other components. These changes also offer much more flexibility in the use of the `<ino-icon>` as you can now utilize the whole API of the `<ino-icon>` component (e.g. using your own icons, providing your custom click handler, ...).

### Features

* **elements|ino-datepicker:** validate input based on date format ([#50](https://github.com/inovex/elements/issues/50)) ([ea4d288](https://github.com/inovex/elements/commit/ea4d2887f9d96777bf294ea66af8766511b7c505))
* **elements|ino-dialog:** add dialog component ([da53a5a](https://github.com/inovex/elements/commit/da53a5a9c9f57a7fb7be91ced04c8a3db8c8ecc7))


### Documentation

* update url's of storybook ([#12](https://github.com/inovex/elements/issues/12)) ([791c8da](https://github.com/inovex/elements/commit/791c8daed3e774b31fb41259bf1410176f63fdaa))


### Miscellaneous

* add heading for refactor commit type ([#54](https://github.com/inovex/elements/issues/54)) ([c5e0bdc](https://github.com/inovex/elements/commit/c5e0bdc793f5a26bd82166475ecd27a17f99f9d9))
* update references to new repository in changelogs ([#16](https://github.com/inovex/elements/issues/16)) ([06f83f1](https://github.com/inovex/elements/commit/06f83f1acd0aef1cf5941766ebc17203648cda52))
* update release workflow ([#13](https://github.com/inovex/elements/issues/13)) ([14b9578](https://github.com/inovex/elements/commit/14b9578fb437b2801a68e478007f581805ea34ef))
* update repository references in packages ([#15](https://github.com/inovex/elements/issues/15)) ([2b38cc5](https://github.com/inovex/elements/commit/2b38cc51f5c76851f127b9618ed3fda475c351a6))


### Refactoring

* change scope from (at)inovex to (at)inovex.de ([#66](https://github.com/inovex/elements/issues/66)) ([96186c2](https://github.com/inovex/elements/commit/96186c2b6f42eb202acf69cd7e0da6280b831864))
* **elements|ino-button:** replace icon property with icon slot ([#28](https://github.com/inovex/elements/issues/28)) ([388ce17](https://github.com/inovex/elements/commit/388ce17df09e8adb8b9d2c7e3a8a7025223cb5f6))

# [0.21.0](https://github.com/inovex/elements/compare/v0.20.0...v0.21.0) (2020-07-02)



# [0.20.0](https://github.com/inovex/elements/compare/v0.19.1...v0.20.0) (2020-06-25)


### Features

* **elements:** add ino-img-list component ([3c15e98](https://github.com/inovex/elements/commit/3c15e9818a46e1f5860eed8e83ad6f200262f71f))



## [0.19.1](https://github.com/inovex/elements/compare/v0.19.0...v0.19.1) (2020-06-18)



# [0.19.0](https://github.com/inovex/elements/compare/v0.18.0...v0.19.0) (2020-06-18)


### Features

* **elements:** add ino-table/-row/-cell components ([fc2f778](https://github.com/inovex/elements/commit/fc2f778ada0c4153d9f67a2a8daef4250228b64f))
* **elements:** add progress bar ([5fbd35e](https://github.com/inovex/elements/commit/5fbd35e9c1a8e2494bdbce4f0148e39d7794d8c0))
* **elements|ino-icon:** ability to set svg title ([e6a6b60](https://github.com/inovex/elements/commit/e6a6b608168364186a840a1952f89afe046f4512))



# [0.18.0](https://github.com/inovex/elements/compare/v0.17.0...v0.18.0) (2020-05-20)


### Features

* **elements|ino-button:** add css variables ([9295b65](https://github.com/inovex/elements/commit/9295b6510029e4c37ef25e535467d1ad3ce90fb2))
* **elements|ino-input:** add methods to focus and blur the input element ([9c9e863](https://github.com/inovex/elements/commit/9c9e863b2d0fa798ab0484132855977a1634df03))
* **elements|ino-option-group:** add ino-option-group component ([6aa7a3c](https://github.com/inovex/elements/commit/6aa7a3ca7bbe771b269b27934bea88ce021b9bc6))



# [0.17.0](https://github.com/inovex/elements/compare/v0.16.0...v0.17.0) (2020-04-29)


### Features

* **elements|ino-checkbox:** add selection variant ([3e74a78](https://github.com/inovex/elements/commit/3e74a7892bdaa9dd12e1be8cf60780881a7b10d4))
* **elements|ino-icon:** unify icons & add styling options with CSS variables ([a755e69](https://github.com/inovex/elements/commit/a755e69a656b591d869f23af818bc8a51b3718cb))
* **elements|ino-nav-drawer:** create nav-drawer-element with nav-item ([72dd914](https://github.com/inovex/elements/commit/72dd9143c3859a9b173a779de4d1ef9114d3f65c))
* **elements|ino-sidebar:** add ino-sidebar ([e9d56de](https://github.com/inovex/elements/commit/e9d56de9abfec70139f8f1cce9f164b245040f89))
* **storybook:** add pure js example and cleanup docs ([7cc9fdb](https://github.com/inovex/elements/commit/7cc9fdb77bf746096dcedfc769f66ecdf981d3ea))



# [0.16.0](https://github.com/inovex/elements/compare/v0.15.1...v0.16.0) (2020-04-01)



## [0.15.1](https://github.com/inovex/elements/compare/v0.15.0...v0.15.1) (2020-03-26)


### Bug Fixes

* **elements:** revert @stencil/core to 1.10.3 due to style issues ([376c79d](https://github.com/inovex/elements/commit/376c79d94d45c4ab52bbf635bbfa9c446a0d7264))



# [0.15.0](https://github.com/inovex/elements/compare/v0.14.0...v0.15.0) (2020-03-25)


### Features

* **elements|ino-button:** add property ino-loading to display a loading spinner ([7cbbce7](https://github.com/inovex/elements/commit/7cbbce7058162e2a585ce08431d28e0113a19cf8))
* **elements|ino-header:** add component ([4c5c69f](https://github.com/inovex/elements/commit/4c5c69f5f97f43beb462197aad0c62ffab2e80b6))
* **elements|ino-segment-group:** add ino-segment-group and -button ([e26478c](https://github.com/inovex/elements/commit/e26478cf593052c23cd707ecb678997dd5cd6e85))



# [0.14.0](https://github.com/inovex/elements/compare/v0.13.0...v0.14.0) (2020-03-17)


### Bug Fixes

* **elements:** fix wrong index in activeTabChange event in the ino-tab-bar ([4c90881](https://github.com/inovex/elements/commit/4c908817d62af58fc6442ebd5271c5b099ade997))


### Features

* **elements:** add option to align snackbar to the right ([eb8b8ab](https://github.com/inovex/elements/commit/eb8b8ab39660e0f5fe4ea3a99ae5aa338b37d31a))
* **elements:** ino-fab(-set) in inovex style ([8bd1b32](https://github.com/inovex/elements/commit/8bd1b32458f60ce2c876e005e5cbc1159e269587))



# [0.13.0](https://github.com/inovex/elements/compare/v0.12.0...v0.13.0) (2020-02-19)



# [0.12.0](https://github.com/inovex/elements/compare/v0.11.0...v0.12.0) (2020-02-12)


### Bug Fixes

* **ino-input:** adjust spacing of ino-icon ([1f796a6](https://github.com/inovex/elements/commit/1f796a64bcd69d1dba450127589174fb1f3d0562))


### Features

* **elements:** add attribute ino-show-label-hint to ino-input component ([6ee24de](https://github.com/inovex/elements/commit/6ee24de3e8953967e28483a7cc2c83e136a5d0bc))
* **elements:** add ino-switch ([23a5512](https://github.com/inovex/elements/commit/23a55126912e218b47170b9a0442b1c2924cc3d7))



# [0.11.0](https://github.com/inovex/elements/compare/v0.10.1...v0.11.0) (2020-02-07)


### Features

* **elements:** add new ino-error prop to ino-input ([3e4ae46](https://github.com/inovex/elements/commit/3e4ae46b5cc26f57dd0c43ffac3b1d1d47d1d626))
* **elements:** add thousands sep., decimal places, unit to ino-input ([a7df4cb](https://github.com/inovex/elements/commit/a7df4cbe13e12c8069fddf29e760df080ccbcfd6))



## [0.10.1](https://github.com/inovex/elements/compare/v0.10.0...v0.10.1) (2019-12-09)



# [0.10.0](https://github.com/inovex/elements/compare/v0.9.0...v0.10.0) (2019-12-09)
