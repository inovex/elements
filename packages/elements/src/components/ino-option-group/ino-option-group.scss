@use '../styles/mdc-customize';
@use "@material/list/mdc-list";

ino-option-group ino-option:not(#ino-opt-group-header) .mdc-list-item {
  padding-left: 30px;
}
