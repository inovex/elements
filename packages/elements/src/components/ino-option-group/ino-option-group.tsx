import { Component, Host, Prop, h } from '@stencil/core';

/**
 * @slot default - One or more `ino-option`
 */
@Component({
  tag: 'ino-option-group',
  styleUrl: 'ino-option-group.scss',
  shadow: false,
})
export class InoOptionGroup {
  /**
   * Label of the group. If not set, this component serves as a wrapper component for dynamically added `ino-options`.
   * When using react and vue, an issue exists with slots and the virtual DOM. Read more about it [here](https://github.com/ionic-team/stencil/issues/2259).
   */
  @Prop() label?: string;

  render() {
    return (
      <Host>
        {this.label && (
          <ino-option id={'ino-opt-group-header'} value={this.label} disabled>
            {this.label}
          </ino-option>
        )}
        <slot></slot>
      </Host>
    );
  }
}
