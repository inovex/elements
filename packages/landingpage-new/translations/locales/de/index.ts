import learn from './learn';
import home from './home';
import about from './about';

export default {
  learn,
  home,
  about,
};
