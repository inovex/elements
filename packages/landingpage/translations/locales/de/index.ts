import about from './about.json';
import common from './common.json';
import explore from './explore.json';
import home from './home.json';
import library from './library.json';

export default {
  home,
  about,
  common,
  library,
  explore,
};
