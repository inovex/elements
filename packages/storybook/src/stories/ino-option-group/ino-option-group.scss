.story-option-group {
  ino-select {
    margin-bottom: 30px;
  }

  margin-bottom: 300px;
}

.story-option-group-customizable {
  margin-bottom: 150px;
}
