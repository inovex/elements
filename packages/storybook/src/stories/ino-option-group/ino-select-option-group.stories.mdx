import { Meta, Description } from "@storybook/addon-docs";
import readme from "../../../../elements/src/components/ino-option-group/readme.md";

<Meta title="Input/ino-option-group/Documentation" />

<Description markdown={readme} />
