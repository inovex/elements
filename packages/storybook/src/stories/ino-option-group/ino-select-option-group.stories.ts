import { html } from 'lit-html';
import { defaultDecorator } from '../utils';
import './ino-option-group.scss';

export default {
  title: 'Input/ino-option-group',
  component: 'ino-option-group',
  decorators: [defaultDecorator],
};

export const Playground = (args) => html`
  <div class="story-option-group-customizable">
    <ino-select label="Select with Group">
      <ino-option-group label="${args.label}">
          <ino-option value="Option 1">Option 1</ino-option>
          <ino-option value="Option 2">Option 2</ino-option>
          <ino-option value="Option 3">Option 3</ino-option>
      </ino-option-group>
    </ino-select>
  </div>
`;
Playground.args = {
  label: 'Group Label',
}

export const MultipleGroups = (args) => html`
  <div class="story-option-group">
    <ino-select label="Select with Groups">
      <ino-option-group label="My First Group">
          <ino-option value="Option 1">Option 1</ino-option>
          <ino-option value="Option 2">Option 2</ino-option>
          <ino-option value="Option 3">Option 3</ino-option>
      </ino-option-group>
      <ino-option-group label="My Second Group">
          <ino-option value="Option 5">Option 5</ino-option>
          <ino-option value="Option 6">Option 6</ino-option>
          <ino-option value="Option 7">Option 7</ino-option>
      </ino-option-group>
    </ino-select>
  </div>
`;
