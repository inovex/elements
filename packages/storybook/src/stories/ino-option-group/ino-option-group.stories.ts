import { Meta } from '@storybook/web-components';
import { Components } from '@inovex.de/elements';
import { html } from 'lit-html';
import Story from '../StoryWrapper';
import './ino-option-group.scss';

const InoOptionGroupMeta = {
  title: 'Input/ino-option-group',
  component: 'ino-option-group',
  parameters: {
    docs: {
      story: {
        height: '450px',
      },
    },
  },
  render: (args) => html`
    <ino-select label="Select with Group">
      <ino-option-group label="${args.label}">
        <ino-option value="Option 1">Option 1</ino-option>
        <ino-option value="Option 2">Option 2</ino-option>
        <ino-option value="Option 3">Option 3</ino-option>
      </ino-option-group>
      <ino-option-group label="${args.label}">
        <ino-option value="Option 4">Option 4</ino-option>
        <ino-option value="Option 5">Option 5</ino-option>
        <ino-option value="Option 6">Option 6</ino-option>
      </ino-option-group>
    </ino-select>
  `,
  args: {
    label: 'My group Label',
  },
} as Meta<Components.InoOptionGroup>;

export default InoOptionGroupMeta;

export const Default = Story({
  ...InoOptionGroupMeta,
});
