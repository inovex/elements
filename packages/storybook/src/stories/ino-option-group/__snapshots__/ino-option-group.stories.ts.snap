// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input/ino-option-group Playground smoke-test 1`] = `
<div id="root-inner">
  <div class="story-option-group">
    <div class="story-option-group-customizable">
      <ino-select label="Select with Group"
                  class="hydrated"
      >
        <div class="mdc-select mdc-select--filled">
          <div class="mdc-select__anchor mdc-ripple-upgraded"
               tabindex="0"
               aria-disabled="false"
               aria-invalid="false"
          >
            <div class="mdc-select__selected-text">
            </div>
            <span class="mdc-select__dropdown-icon">
              <svg class="mdc-select__dropdown-icon-graphic"
                   viewbox="7 10 10 5"
              >
                <polygon class="mdc-select__dropdown-icon-inactive"
                         stroke="none"
                         fill-rule="evenodd"
                         points="7 10 12 15 17 10"
                >
                </polygon>
                <polygon class="mdc-select__dropdown-icon-active"
                         stroke="none"
                         fill-rule="evenodd"
                         points="7 15 12 10 17 15"
                >
                </polygon>
              </svg>
            </span>
            <ino-label class="hydrated">
              <div class="mdc-line-ripple">
              </div>
              <label class="mdc-floating-label">
                Select with Group
              </label>
            </ino-label>
          </div>
          <div class="mdc-select__menu mdc-menu mdc-menu-surface mdc-menu-surface--fullwidth">
            <ul class="mdc-deprecated-list">
              <ino-option-group label="My group Label"
                                class="hydrated"
              >
                <ino-option id="ino-opt-group-header"
                            class="hydrated"
                >
                  <li class="mdc-deprecated-list-item mdc-deprecated-list-item--disabled"
                      data-value="My group Label"
                      tabindex="-1"
                  >
                    <span class="mdc-deprecated-list-item__text">
                      My group Label
                    </span>
                  </li>
                </ino-option>
                <ino-option value="Option 1"
                            class="hydrated"
                >
                  <li class="mdc-deprecated-list-item"
                      data-value="Option 1"
                      tabindex="-1"
                  >
                    <span class="mdc-deprecated-list-item__text">
                      Option 1
                    </span>
                  </li>
                </ino-option>
                <ino-option value="Option 2"
                            class="hydrated"
                >
                  <li class="mdc-deprecated-list-item"
                      data-value="Option 2"
                      tabindex="-1"
                  >
                    <span class="mdc-deprecated-list-item__text">
                      Option 2
                    </span>
                  </li>
                </ino-option>
                <ino-option value="Option 3"
                            class="hydrated"
                >
                  <li class="mdc-deprecated-list-item"
                      data-value="Option 3"
                      tabindex="-1"
                  >
                    <span class="mdc-deprecated-list-item__text">
                      Option 3
                    </span>
                  </li>
                </ino-option>
              </ino-option-group>
              <ino-option-group label="My group Label"
                                class="hydrated"
              >
                <ino-option id="ino-opt-group-header"
                            class="hydrated"
                >
                  <li class="mdc-deprecated-list-item mdc-deprecated-list-item--disabled"
                      data-value="My group Label"
                      tabindex="-1"
                  >
                    <span class="mdc-deprecated-list-item__text">
                      My group Label
                    </span>
                  </li>
                </ino-option>
                <ino-option value="Option 4"
                            class="hydrated"
                >
                  <li class="mdc-deprecated-list-item"
                      data-value="Option 4"
                      tabindex="-1"
                  >
                    <span class="mdc-deprecated-list-item__text">
                      Option 4
                    </span>
                  </li>
                </ino-option>
                <ino-option value="Option 5"
                            class="hydrated"
                >
                  <li class="mdc-deprecated-list-item"
                      data-value="Option 5"
                      tabindex="-1"
                  >
                    <span class="mdc-deprecated-list-item__text">
                      Option 5
                    </span>
                  </li>
                </ino-option>
                <ino-option value="Option 6"
                            class="hydrated"
                >
                  <li class="mdc-deprecated-list-item"
                      data-value="Option 6"
                      tabindex="-1"
                  >
                    <span class="mdc-deprecated-list-item__text">
                      Option 6
                    </span>
                  </li>
                </ino-option>
              </ino-option-group>
            </ul>
          </div>
        </div>
      </ino-select>
    </div>
  </div>
</div>
`;
